src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -371,9 +371,31 @@ private void UpdateInputScope(InputScope inputScope)
 
 		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
-			//TODO : see bug #8178
+			if (e.NewValue is TextWrapping textWrapping && !string.IsNullOrEmpty(Text) &&
+				(textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords))
+			{
+
+				if (_textBoxView != null)
+				{
+					InternalIsSingleLine = _textBoxView.IsSingleLine;
+
+					if (_textBoxView.IsSingleLine)
+					{
+						_textBoxView.SetSingleLine(false);
+					}
+				}
+			}
+			else
+			{
+				if (InternalIsSingleLine.HasValue)
+				{
+					_textBoxView?.SetSingleLine(InternalIsSingleLine.Value);
+				}
+			}
 		}
 
+		private bool? InternalIsSingleLine { get; set; }
```

Comment:
This doesn't need to be a property. It can be a field.

```suggestion
		private bool? _isSingleLine;
```

It's unclear to me what the `null` value means, when it's set and it's never reset.